src/scripts/analyze_missingness.py

```diff
@@ -67,6 +67,30 @@ def count_records_missing_data(
     return results
 
 
+def find_one_patient_per_report(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    ECR and VXU data sets, as they've been given to us, contain a
+    patient's entire case or vaccination history, respectively. This
+    means when we convert to FHIR, all individual observations are split
+    into separate messages, which the CSV generator parses as separate
+    rows. The effect is that one patient could appear multiple times
+    in a CSV for covid-related reasons, even if in the data they formed
+    a single concrete unit. This function is a quick and dirty hack
+    to work around this by preserving only the first instance of a
+    patient's hash in the data set. It is useful PURELY for determining
+    how many records our hash actually linked.
+    """
+    prev_hash = None
+    idxs_to_keep = []
+    for row in df.index:
+        if prev_hash == None or df["patientHash"][row] != prev_hash:
+            idxs_to_keep.append(row)
+            prev_hash = df["patientHash"][row]
+    new_df = df.loc[idxs_to_keep, :]
+    new_df.reset_index()
```

Comment:
Do we need to add a `drop=True` to prevent the old index from getting added as a variable?

Reply:
We can. It doesn't really change the result since we drop all the columns that aren't equity fields anyway, but for clarity and cleanliness yeah we probably should.